build.rs

```diff
@@ -574,10 +689,27 @@ fn main() -> Result<()> {
     // If you have troubles with your shell accepting '.' in a var name,
     // try using 'env' (sorry but this isn't our fault - it just has to
     // match the pkg-config package name, which is going to have a . in it).
-    let cross_compiling =
-        env::var("PYO3_CROSS_INCLUDE_DIR").is_ok() && env::var("PYO3_CROSS_LIB_DIR").is_ok();
+    //
+    // Detecting if cross-compiling by checking if the target triple is different from the host
+    // rustc's triple.
+    let cross_compiling = env::var("TARGET") != env::var("HOST");
     let (interpreter_config, mut config_map) = if cross_compiling {
-        load_cross_compile_info()?
+        // If cross compiling we need the path to the cross-compiled include dir and lib dir, else
+        // fail quickly and loudly
+        let python_include_dir = match env::var("PYO3_CROSS_INCLUDE_DIR") {
+            Ok(v) => v,
+            Err(_) => bail!(
+                "Must provide PYO3_CROSS_INCLUDE_DIR environment variable when cross-compiling"
+            ),
+        };
```

Comment:
Maybe if we drop the header parsing then we can also drop this env var? Which might be nice.

Reply:
While I havn't delved into the actual library code, don't you need the include directory for the FFI functions? If not then removing this variable would in fact improve cross compilation ergonomics

Reply:
I've dropped it for unix platforms, unfortunately, windows won't have this information